python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -443,6 +457,14 @@ def format(self, ref):
             if field_name in ("run", "collection"):
                 usedRunOrCollection = True
 
+            # Check for request for additional information from the dataId
+            if "." in field_name:
+                primary, secondary = field_name.split(".")
+                if primary in extras:
+                    element = extras[primary]
+                    if hasattr(element, secondary):
+                        fields[field_name] = getattr(element, secondary)
```

Comment:
What happens if the template has a dotted name that doesn't exist?  From this it looks like we just don't do anything with it, but I'm probably not seeing the big picture.

Reply:
A few lines later you get an exception because `field_name` won't be in `fields`. It will say something like "detector.name is not defined" or something.